src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -59488,6 +59520,2310 @@ class C {
+                // (3,19): error CS0265: Partial declarations of 'I1<TF1>' have inconsistent constraints for type parameter 'TF1'
+                // partial interface I1<TF1> where TF1 : object?, new()
+                Diagnostic(ErrorCode.ERR_PartialWrongConstraints, "I1").WithArguments("I1<TF1>", "TF1").WithLocation(3, 19),
+                // (3,39): error CS0702: Constraint cannot be special class 'object?'
+                // partial interface I1<TF1> where TF1 : object?, new()
+                Diagnostic(ErrorCode.ERR_SpecialTypeAsBound, "object?").WithArguments("object?").WithLocation(3, 39)
+                );
+        }
+
+        [Fact]
+        public void Constraints_128()
+        {
+            var source1 =
+@"
+partial class A
+{
+#nullable disable
+partial void M1<TF1>();
+
+partial void M1<TF1>()
+{
+}
+}
+";
+            var comp1 = CreateCompilation(source1);
+
+            comp1.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void Constraints_129()
+        {
+            var source1 =
+@"
+partial class A
+{
+#nullable enable
+partial void M1<TF1>();
+
+partial void M1<TF1>()
+{
+}
+}
+";
+            var comp1 = CreateCompilation(source1);
+
+            comp1.VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void Constraints_130()
+        {
+            var source1 =
+@"
+partial class A
+{
+#nullable disable
+partial void M1<TF1>();
+
+#nullable enable
+partial void M1<TF1>()
+{
+}
+}
+";
+            var comp1 = CreateCompilation(source1);
+
+            comp1.VerifyDiagnostics(
+                // (8,14): error CS0761: Partial method declarations of 'A.M1<TF1>()' have inconsistent type parameter constraints
+                // partial void M1<TF1>()
+                Diagnostic(ErrorCode.ERR_PartialMethodInconsistentConstraints, "M1").WithArguments("A.M1<TF1>()").WithLocation(8, 14)
+                );
+
+            var source2 =
+@"
+partial class A
+{
+#nullable enable
+partial void M1<TF1>()
+{
+}
+
+#nullable disable
+partial void M1<TF1>();
+}
+";
+            var comp2 = CreateCompilation(source2);
+
+            comp2.VerifyDiagnostics(
+                // (5,14): error CS0761: Partial method declarations of 'A.M1<TF1>()' have inconsistent type parameter constraints
+                // partial void M1<TF1>()
+                Diagnostic(ErrorCode.ERR_PartialMethodInconsistentConstraints, "M1").WithArguments("A.M1<TF1>()").WithLocation(5, 14)
+                );
+        }
+
+        [Fact]
+        public void Constraints_131()
+        {
+            var source1 =
+@"
+partial class A
+{
+#nullable enable
+partial void M1<TF1>();
+
+#nullable disable
+partial void M1<TF1>()
+{
+}
+}
+";
+            var comp1 = CreateCompilation(source1);
+
+            comp1.VerifyDiagnostics(
+                // (8,14): error CS0761: Partial method declarations of 'A.M1<TF1>()' have inconsistent type parameter constraints
+                // partial void M1<TF1>()
```

Comment:
By LDM decisions [https://github.com/dotnet/csharplang/blob/master/meetings/2017/LDM-2017-08-09.md#unconstrained-generics](https://github.com/dotnet/csharplang/blob/master/meetings/2017/LDM-2017-08-09.md#unconstrained-generics) these have identical constraints and there should be no diagnostic.